src/controllers/selectDirector.controller.ts

```diff
@@ -0,0 +1,105 @@
+import ValidationErrors from 'app/models/validationErrors'
+import { CHECK_YOUR_ANSWERS_URI, DEFINE_SIGNATORY_INFO_URI, SELECT_DIRECTOR_URI, SELECT_SIGNATORIES_URI } from 'app/paths'
+import selectDirectorSchema from 'app/schemas/selectDirector.schema'
+import CompanyOfficersService from 'app/services/company-officers/companyOfficers.service'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
+import FormValidator from 'app/utils/formValidator.util'
+
+interface ViewModel {
+  directors: DirectorDetails[]
+  data?: Optional<SelectDirectorFormModel>
+  errors?: Optional<ValidationErrors>
+}
+
+// TODO - include company auth middleware when available
+@controller(SELECT_DIRECTOR_URI, TYPES.SessionMiddleware, TYPES.AuthMiddleware)
+export class SelectDirectorController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(CompanyOfficersService) private officerService: CompanyOfficersService,
+    @inject(FormValidator) private validator: FormValidator) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<string> {
+    const form: Optional<SelectDirectorFormModel> = this.getFormFromSession()
+
+    const directors: DirectorDetails[] = await this.getDirectors()
+
+    return this.renderView(directors, form)
+  }
+
+  private getFormFromSession(): Optional<SelectDirectorFormModel> {
+    return this.session.getDissolutionSession(this.httpContext.request)!.selectDirectorForm
+  }
+
+  @httpPost('')
+  public async post(@requestBody() body: SelectDirectorFormModel): Promise<string | void> {
+    const directors: DirectorDetails[] = await this.getDirectors()
+
+    const errors: Optional<ValidationErrors> = this.validator.validate(body, selectDirectorSchema)
+    if (errors) {
+      return this.renderView(directors, body, errors)
+    }
+
+    this.updateSession(body)
+
+    return this.httpContext.response.redirect(this.getRedirectURI(directors, body))
+  }
+
+  private async getDirectors(): Promise<DirectorDetails[]> {
+    const companyNumber: string = this.getCompanyNumber()
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+
+    return this.officerService.getActiveDirectorsForCompany(token, companyNumber)
+  }
+
+  private getCompanyNumber(): string {
+    return this.session.getDissolutionSession(this.httpContext.request)!.companyNumber!
+  }
+
+  private async renderView(
+    directors: DirectorDetails[],
+    data?: Optional<SelectDirectorFormModel>,
+    errors?: Optional<ValidationErrors>): Promise<string> {
+    const viewModel: ViewModel = {
+      directors,
+      data,
+      errors
+    }
+
+    return super.render('select-director', viewModel, errors ? BAD_REQUEST : OK)
+  }
+
+  private updateSession(body: SelectDirectorFormModel): void {
+    const updatedSession: DissolutionSession = {
+      ...this.session.getDissolutionSession(this.httpContext.request),
```

Comment:
For those unfamiliar, this is the Object Spread syntax, which enables fluent extensions to objects.